android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java

```diff
@@ -3,18 +3,134 @@
 import android.os.Bundle;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
 
+import com.chorestory.Interface.RetrofitInterface;
 import com.chorestory.R;
+import com.chorestory.adapter.ChildRecyclerViewAdapter;
+import com.chorestory.app.App;
+import com.chorestory.helpers.Toaster;
+import com.chorestory.helpers.TokenHandler;
+import com.chorestory.templates.AddFriendResponse;
+import com.chorestory.templates.ClanChildrenResponse;
+import com.chorestory.templates.ClanChildrenResponse.Children;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class ChildFriendsFragment extends ChoreStoryFragment {
+    @Inject
+    RetrofitInterface retrofitInference;
+    @Inject
+    TokenHandler tokenHandler;
+
+    private List<Children> childrenList;
+    private RecyclerView childRecyclerView;
+    private RecyclerView.LayoutManager childLayoutManager;
+    private RecyclerView.Adapter childAdapter;
+    private Button addFriendButton;
+    private EditText addFriendEditText;
 
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_child_friends, container, false);
 
+        App.getAppComponent().inject(this);
+
+        childRecyclerView = view.findViewById(R.id.child_recycler_view);
+        addFriendButton = view.findViewById(R.id.add_friend_button);
+        addFriendEditText = view.findViewById(R.id.add_friend_edit_text);
+        childRecyclerView.setHasFixedSize(true);
+
+        childLayoutManager = new LinearLayoutManager(getActivity());
+        childRecyclerView.setLayoutManager(childLayoutManager);
+
+
+        addFriendButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String username = addFriendEditText.getText().toString();
```

Comment:
add `disableViews()` at the start on onClick
and add `enableViews()`in the appropriate places after